lib/ace/mode/r_highlight_rules.js

```diff
@@ -46,15 +46,117 @@ define(function(require, exports, module)
    var RHighlightRules = function()
    {
 
-      var keywords = lang.arrayToMap(
-            ("function|if|in|break|next|repeat|else|for|return|switch|while|try|tryCatch|stop|warning|require|library|attach|detach|source|setMethod|setGeneric|setGroupGeneric|setClass")
-                  .split("|")
-            );
+      var keywords = ( // from R's ls('package:base')
+          "function|if|in|break|next|repeat|else|for|return|switch|while|try|tryCatch|stop|" +
+          "warning|require|library|attach|detach|source|setMethod|setGeneric|setGroupGeneric|setClass|" +
+          "Arg|Cstack_info|Encoding<-|Filter|I|ISOdatetime|LETTERS|La_version|Math.Date|" +
+          "Math.data.frame|Math.factor|NCOL|Negate|OlsonNames|Ops.POSIXt|Ops.difftime|" +
+          "Ops.numeric_version|Position|R.home|R.version.string|RNGversion|Re|Reduce|Summary.POSIXct|" +
+          "Summary.data.frame|Summary.factor|Summary.ordered|Sys.chmod|Sys.getlocale|Sys.glob|" +
+          "Sys.localeconv|Sys.setFileTime|Sys.setlocale|Sys.time|Sys.umask|Sys.which|UseMethod|" +
+          "[.Date|[.POSIXct|[.data.frame|[.factor|[.listof|[.numeric_version|[.simple.list|[<-|" +
```

Comment:
keywords with `[` and `<-` won't be matched because regex on the token doesn't allow them, and <- looks like a separate operator.

Are things like Sys.time one keyword or keyword+operator+keyword?